File: pdlearning/src/microservices/course-microservice/Api/Application/Models/AssessmentAnswerModel.cs
using System;
using System.Collections.Generic;
using Microservice.Course.Domain.Entities;

namespace Microservice.Course.Application.Models
{
    public class AssessmentAnswerModel
    {
        public AssessmentAnswerModel(AssessmentAnswer entity)
        {
            Id = entity.Id;
            AssessmentId = entity.AssessmentId;
            ParticipantAssignmentTrackId = entity.ParticipantAssignmentTrackId;
            UserId = entity.UserId;
            CriteriaAnswers = entity.CriteriaAnswers;
            CreatedBy = entity.CreatedBy;
            ChangedBy = entity.ChangedBy;
            SubmittedDate = entity.SubmittedDate;
            ChangedDate = entity.ChangedDate;
        }

        public Guid Id { get; set; }

        public Guid AssessmentId { get; set; }

        public Guid ParticipantAssignmentTrackId { get; set; }

        public Guid UserId { get; set; }

        public IEnumerable<AssessmentCriteriaAnswer> CriteriaAnswers { get; set; }

        public DateTime? SubmittedDate { get; set; }

        public Guid CreatedBy { get; set; }

        public Guid ChangedBy { get; set; }

        public DateTime? ChangedDate { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/assessment-answer.model.ts
import { AssessmentCriteriaAnswer, IAssessmentCriteriaAnswer } from './assessment-criteria-answer.model';

export interface IAssessmentAnswer {
  id?: string;
  assessmentId: string;
  participantAssignmentTrackId: string;
  userId: string;
  submittedDate?: Date;
  changedBy: string;
  changedDate?: Date;
  criteriaAnswers: IAssessmentCriteriaAnswer[];
}

export class AssessmentAnswer implements IAssessmentAnswer {
  public static assessmentForFacilitator = '00000000-0000-0000-0000-000000000000';
  public id?: string;
  public assessmentId: string;
  public participantAssignmentTrackId: string;
  public userId: string;
  public submittedDate?: Date;
  public changedBy: string;
  public changedDate?: Date;
  public criteriaAnswers: AssessmentCriteriaAnswer[] = [];

  constructor(data?: IAssessmentAnswer) {
    if (data) {
      this.id = data.id;
      this.assessmentId = data.assessmentId;
      this.participantAssignmentTrackId = data.participantAssignmentTrackId;
      this.userId = data.userId;
      this.submittedDate = data.submittedDate;
      this.changedBy = data.changedBy;
      this.criteriaAnswers = data.criteriaAnswers != null ? data.criteriaAnswers.map(x => new AssessmentCriteriaAnswer(x)) : [];
      this.changedDate = data.changedDate;
    }
  }

  public isAssessentForFacilitator(): boolean {
    return this.userId === AssessmentAnswer.assessmentForFacilitator;
  }
}
